source/parse.h

```diff
@@ -1913,20 +1913,10 @@ struct compound_statement_node
         return open_brace;
     }
 
-    auto add_statement( 
+    auto add_statement(
         std::unique_ptr<statement_node>&& statement,
         int                               before_pos
-    )
-        -> bool
-    {
-        //  Adopt this statement into our list of statements
-        statements.insert( 
```

Comment:
[Explanation] This call causes deletion of the `std::unique_ptr<statement_node>`. As discussed in #1088  this causes trouble for clang-18 if `compound_statement_node::add_statement()` is defiend before `statement_node::~statement_node()`.